metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -378,34 +397,28 @@ void build_pred_succ(const std::string &graph_fname,
         std::vector<bool> pred_boundary_buf;
 
         for (uint64_t i = start; i < std::min(start + BS, graph.num_nodes() + 1); ++i) {
-            BOSS::edge_index boss_idx = graph.kmer_to_boss_index(i);
-            if (!dummy[boss_idx]) {
+            bool find_succ = true;
+            if (succinct) { // Legacy code for DBGSuccinct
+                const BOSS &boss = succinct->get_boss();
+                BOSS::edge_index boss_idx = succinct->kmer_to_boss_index(i);
```

Comment:
I removed the check that `boss_idx` is not dummy, because, from my perspective, nodes from `1` to `num_nodes()` should never be mapped into a dummy node and if they do, it probably should be treated as a misbehavior of `DBGSuccinct`? Note that it never seemed to actually be a dummy node in `LabeledAlignerTest` test group.